src/state_manager/mod.rs

```diff
@@ -247,6 +266,11 @@ where
             beacon,
             chain_config,
             sync_config,
+            enable_event_caching: if enable_event_caching {
+                EventCache::Cached
+            } else {
+                EventCache::NotCached
+            },
```

Comment:
Design idea - I wonder if we can make the code more straightforward by injecting an trait object of `EventCache`. It would be instantiated at the start of the program to either an implementation without cache or with a cache. This way, throughout the code, you wouldn't need to check if the caching is enabled or not - you'd write to it but whether it would actually happen would be completely transparent.

What do you think?

Reply:
Excellent idea! In the future we'll likely have:
- No cache (e.g., a node without RPC)
- Cache Events in tipset state cache
- Use sidecar indexer to retrieve Events

Reply:
I've updated the code to store events in a dedicated cache (with a dedicated size), avoiding the need to store unnecessary `None` values in the SM tipset state cache when events are not in use. This also prevents polluting the SM cache and avoids potential performance slowdowns.

As for injecting an trait object, I think it would add some complexity compared to using an `enum` at this stage. Let's consider that for a future PR, or once we have an indexer in place?

Reply:
One more thing, we can remove the `--enable-event-caching` flag now.